chia/wallet/wallet_node.py

```diff
@@ -1086,37 +1101,62 @@ async def fetch_and_validate_the_weight_proof(
     ) -> Tuple[bool, Optional[WeightProof], List[SubEpochSummary], List[BlockRecord]]:
         assert self.wallet_state_manager is not None
         assert self.wallet_state_manager.weight_proof_handler is not None
+        capabilities = peer.capabilities
```

Comment:
same comments as full node.
Also the code here is very similar, maybe we can merge it?